apps/api/src/domain/repositories/interfaces/ICategoryRepository.ts

```diff
@@ -0,0 +1,12 @@
+import { InternalError } from '../../../lib/errors/InternalError';
+import { Maybe } from '../../../lib/types/maybe';
+import { Category } from '../../models/categories/entities/category.entity';
+import { EntityId } from '../../types/ResourceId';
+
+export interface ICategoryRepository {
+    fetchTree: () => Promise<(Category | InternalError)[]>;
```

Comment:
I like this, more explicit than fetchMany for the categories

Reply:
I think it is important to emphasize that you are dealing with a tree and not just an ordinary array of n elements that can be separated without respecting their relationships.